### PR TITLE
Docs/secrets sync ssrf private endpoints

### PR DESCRIPTION
**Pull request by pixambi**
🔍 [Deploy preview](https://vault-git-docs-secrets-sync-ssrf-private-endpoints-hashicorp.vercel.app/vault/docs/upgrading/upgrade-to-1.17.x#secrets-sync-ssrf-protection-may-block-private-endpoints)

The Secrets Sync update in 1.17.3 introduced SSRF protection that blocks private IP ranges, affecting users accessing secret stores through private endpoints. Adding documentation about the change

### Description
This PR adds documentation to the 1.17.x upgrade guide about a breaking change in Secrets Sync introduced in version 1.17.3. The documentation:

- Explains the new SSRF protection mechanism added to Secrets Sync
- Provides example error messages users might encounter
- Lists current options for users affected by this change

This change helps users understand why their Secrets Sync operations might fail after upgrading to 1.17.3 and what options are available to them.

### TODO only if you're a HashiCorp employee
- [ ] **Backport Labels:** If this PR is in the ENT repo and needs to be backported, backport  
  to N, N-1, and N-2, using the `backport/ent/x.x.x+ent` labels. If this PR is in the CE repo, you should only backport to N, using the `backport/x.x.x` label, not the enterprise labels.
    - [ ] If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, *double check* that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, *not* in your ENT PR.

### CODE REVIEW

File: website/content/partials/known-issues/1_17_secrets-sync-ssrf-private-endpoints.mdx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+As of version 1.17.3, Vault's Secrets Sync includes additional Server-Side Request Forgery (SSRF) protection measures. This security enhancement prevents sync operations to certain IP ranges by introducing a new SSRF-safe HTTP client. The client specifically blocks requests to private IP ranges (such as 10.0.0.0/8), which affects users accessing cloud provider secret stores through private endpoints.
+
+**Impact:**
+- Secrets Sync operations to private IP ranges will be blocked
+- Affects all destinations when accessed via private endpoints
+
+**Example error message:**
+```
+couldn't sync secret with store: failed to publish event: dial tcp [IP]: prohibited IP address: [IP] is not a permitted destination (denied by: 10.0.0.0/8)
+```
```
Comment (yhyakuna) [inline website/content/partials/known-issues/1_17_secrets-sync-ssrf-private-endpoints.mdx:12]: Disable copy to clipboard button.

````suggestion
**Example error message:**

<CodeBlockConfig hideClipboard>

```
couldn't sync secret with store: failed to publish event: dial tcp [IP]: prohibited IP address: [IP] is not a permitted destination (denied by: 10.0.0.0/8)
```

</CodeBlockConfig>
````

**Review by yhyakuna — COMMENTED** [review]
Added minor suggestions.

File: website/content/partials/known-issues/1_17_secrets-sync-ssrf-private-endpoints.mdx (hunk 1 above)
Comment (pixambi) [reply website/content/partials/known-issues/1_17_secrets-sync-ssrf-private-endpoints.mdx:12]: thank you

File: website/content/partials/known-issues/1_17_secrets-sync-ssrf-private-endpoints.mdx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+### Secrets Sync SSRF Protection May Block Private Endpoints
```
[the remark this answers is not in the document]
Comment (pixambi) [reply website/content/partials/known-issues/1_17_secrets-sync-ssrf-private-endpoints.mdx:1]: thank you

**Review by pixambi — COMMENTED** [review]
N/A

File: website/content/partials/known-issues/1_17_secrets-sync-ssrf-private-endpoints.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (pixambi) [reply website/content/partials/known-issues/1_17_secrets-sync-ssrf-private-endpoints.mdx:1]: thank you

**Review by pixambi — COMMENTED** [review]
accepted all changes

**Review by yhyakuna — APPROVED** [review]

**Review by jonathanfrappier — APPROVED** [review]

**Pull request merged**